api-gateway/src/main/java/org/swasth/apigateway/models/BaseRequest.java

```diff
@@ -181,6 +181,11 @@ protected Map<String, ClientException> getResponseParamErrors() {
 
     public void validateUsingAuditData(List<String> allowedEntitiesForForward, List<String> allowedRolesForForward, Map<String, Object> senderDetails, Map<String, Object> recipientDetails, List<Map<String, Object>> correlationAuditData, List<Map<String, Object>> callAuditData, List<Map<String, Object>> participantCtxAuditData, String path) throws Exception {
         validateCondition(!callAuditData.isEmpty(), ErrorCodes.ERR_INVALID_API_CALL_ID, API_CALL_SAME_MSG);
+        //validate correlation id belongs to same cycle or not
+        if(!correlationAuditData.isEmpty() && !correlationAuditData.get(0).get(ACTION).toString().contains(getEntity(apiAction))){
```

Comment:
Why are we checking for only in index zero?

Reply:
All the requests with a correlation id, that are passed through the gateway will contain the same entity(coverageeligibility, preauth, claims etc). So getting first audit event and checking the match of entity names with incoming request.